### DISCUSSION TITLE
Comparable return value of `focus-tab`

### DISCUSSION THREAD

**Discussion by tuxflo** [kovidgoyal/kitty#4376] [Q&A] [answered]
I'm trying to do something like [tmux-sessionizer](https://github.com/ThePrimeagen/.dotfiles/blob/master/bin/.local/bin/tmux-sessionizer) but instead of using tmux, I want to use kitty tabs.

So basically I want to
* open a fuzzy finder (fzf) that lists directories from my workspace folder
* select one
* if a tab with this cwd (or title) exists, jump to this tab
* if the tab doesn't exist: create a new kitty tab with title and cwd option and open a command (nvim) in this tab

I thought the easiest way to implement this would be the use of `kitty @ focus-tab` with the result from the fuzzy finder. However I'm unable to get the return code of the `focus-tab` command when I try to do something like this:

```
if [[ `kitty @ focus-tab --match cwd:/home/me/Documents` -ne 0 ]]; then kitty @ new-window --new-tab --title Documents --cwd /home/me/Documents --no-response; fi
```
If there is no corresponding tab open I get `No matching tabs for expression: cwd:/home/me/Documents` in the terminal as a result, but no return code which I can use to check if the tab exists or not.

(I thought there might be a way do get this working without creating a custom kitten)

**Comment by kovidgoyal** [comment, marked as answer]
If you want to check exit codes of a command in a shell script you do:
```
command
if [[ $? -ne 0 ]]; then ...
```

You are comparing the stdout of a command with zero.

**Reply by tuxflo** [reply]
Thanks a lot for the quick reply!
I had a typo in my command, so that it wouldn't save the return value into `$?` even when I did the command on a single line, without the `if [[ ...` condition.

The only thing I don't know now is: if I append a command, such as `nvim` to the `new-window` call, the tab is closed if neovim is closed (by using `:q`).
Is there an option to keep the tab open, even if the first provided command exits or do I have to switch to the `kitty @ launch` command?

**Comment by kovidgoyal** [comment]
On Mon, Dec 20, 2021 at 04:32:35AM -0800, tuxflo wrote:
> The only thing I don't know now is: if I append a command, such as `nvim` to the `new-window` call, the tab is closed if neovim is closed (by using `:q`).
> Is there an option to keep the tab open, even if the first provided command exits or do I have to switch to the `kitty @ launch` command?


You should switch to launch anyway, its the recommended way. But if you
want to just us sh -c "your command; bash"
